python/cudf/cudf/core/frame.py

```diff
@@ -2280,19 +2321,42 @@ def drop_duplicates(
         if len(subset_cols) == 0:
             return self.copy(deep=True)
 
-        result = self.__class__._from_data(
-            *libcudf.stream_compaction.drop_duplicates(
-                self,
-                keys=subset,
-                keep=keep,
-                nulls_are_equal=nulls_are_equal,
-                ignore_index=ignore_index,
-            )
+        keys = self._positions_from_column_names(
+            subset, include_index=not ignore_index
+        )
+        result = self._drop_duplicates(
+            keys=keys,
+            keep=keep,
+            nulls_are_equal=nulls_are_equal,
+            ignore_index=ignore_index,
         )
-
         result._copy_type_metadata(self)
+        if self._index is not None:
+            result._index.name = self._index.name
+            if isinstance(self._index, cudf.MultiIndex):
+                result._index.names = self._index.names
         return result
 
+    def _positions_from_column_names(self, column_names, include_index=False):
+        """Map each column name into their positions in the frame.
+        """
+        return [
+            i
+            for i, name in enumerate(self._column_names)
+            if name in column_names
+        ]
```

Comment:
I wonder if it would make sense to unify this method with `_column_name_to_indices`? Is it the same intended behavior as calling `_column_name_to_indices` with `include_index=False`?
```suggestion
        return [self._column_names.index(name) for name in column_names]
```

Reply:
Yes, this is an oversight from my end. there should only be one of these functions. And I opted for `_positions_for_column_names`. "Indices" is a very overloaded word in this context and I would rather rephrase it. Let me know if you have better ideas.

Reply:
`_position_from_column_names` also has overloads in `indexed_frame` to actually take `include_index` into account.